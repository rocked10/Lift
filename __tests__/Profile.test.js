import React from 'react';

import Profile from "../screens/profile";
import { render, fireEvent, waitFor } from "@testing-library/react-native";

describe('Profile', () => {
    it('navigates to add post on button press', () => {
        const navigate = jest.fn();
        const { getByTestId } = render(<Profile navigation={{ navigate }} route={false} />);
        fireEvent.press(getByTestId('Edit Profile Button'));
        expect(navigate).toHaveBeenCalledWith('Edit Profile', {
            userProfile: expect.any(Object)
        });
    });
});