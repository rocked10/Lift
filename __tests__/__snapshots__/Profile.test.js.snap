// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Should render 1`] = `
<View
  style={
    Object {
      "padding": 18,
    }
  }
>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "padding": 8,
          }
        }
      >
        <Text />
        <View
          style={
            Object {
              "borderRadius": 60,
              "height": 120,
              "overflow": "hidden",
              "width": 120,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "https://moonvillageassociation.org/wp-content/uploads/2018/06/default-profile-picture1.jpg",
              }
            }
            style={
              Object {
                "flex": 1,
                "height": undefined,
                "width": undefined,
              }
            }
            testID="Profile Picture"
          />
        </View>
        <Text
          style={
            Object {
              "fontFamily": "lato-bold",
              "fontSize": 22,
              "marginTop": 5,
            }
          }
          testID="name"
        />
        <Text
          style={
            Object {
              "fontFamily": "lato-regular",
              "fontSize": 14,
              "marginTop": 5,
            }
          }
          testID="role"
        />
        <Text
          style={
            Object {
              "fontFamily": "lato-regular",
              "fontSize": 16,
              "marginTop": 8,
            }
          }
          testID="bio"
        />
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "padding": 14,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#6200ee",
                "borderColor": "transparent",
                "borderRadius": 10,
                "borderStyle": "solid",
                "borderWidth": 0,
                "elevation": 2,
                "marginHorizontal": 12,
                "minWidth": 64,
                "shadowColor": "#000000",
                "shadowOffset": Object {
                  "height": 0.75,
                  "width": 0,
                },
                "shadowOpacity": 0.24,
                "shadowRadius": 1.5,
              }
            }
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": undefined,
                }
              }
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  Object {
                    "borderRadius": 10,
                  },
                ]
              }
              testID="Edit Profile Button"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "#000000",
                        "fontFamily": "System",
                        "fontWeight": "400",
                      },
                      Object {
                        "textAlign": "left",
                      },
                      Array [
                        Object {
                          "letterSpacing": 1,
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        undefined,
                        Object {
                          "textTransform": "uppercase",
                        },
                        Object {
                          "color": "#ffffff",
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        Object {
                          "fontFamily": "System",
                          "fontWeight": "500",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <Text
                    style={
                      Object {
                        "fontFamily": "karla-bold",
                      }
                    }
                  >
                    EDIT PROFILE
                  </Text>
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#F5F5F5",
            "borderRadius": 16,
            "elevation": 3,
            "marginHorizontal": 2,
            "marginVertical": 9,
            "shadowColor": "#333",
            "shadowOffset": Object {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 2,
          }
        }
      >
        <View
          style={
            Object {
              "marginHorizontal": 18,
              "marginVertical": 10,
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 14,
              }
            }
          >
            <Text />
            <Text
              style={
                Object {
                  "fontFamily": "lato-bold",
                  "fontSize": 22,
                  "marginBottom": 16,
                  "marginHorizontal": 6,
                }
              }
            >
              Personal Records
            </Text>
            <Text />
          </View>
          <Text
            style={
              Object {
                "fontFamily": "lato-regular",
                "fontSize": 18,
                "marginBottom": 16,
              }
            }
          >
            You don't have any yet! Time to get to work...
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
